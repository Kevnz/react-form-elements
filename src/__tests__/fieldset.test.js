import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import {
  Button,
  TextBox,
  CheckBox,
  Range,
  DateTime,
  Telephone,
  DropDown,
  Option,
  OptionGroup,
  Radio,
  Form,
  Fieldset,
} from '../index'

const App = ({ onSubmit }) => {
  return (
    <Form
      name="testForm"
      onSubmit={e => {
        onSubmit(e)
      }}
    >
      <Fieldset legend="My Form">
        <TextBox label="My Label" name="myTextBox" />
        <CheckBox
          label="My Checkbox"
          value="checkbox value"
          isChecked={false}
          name="myCheckBox"
        />
        <Radio
          label="My Radio"
          name="myRadio"
          isChecked={false}
          value="myRadio1"
        />
        <DateTime label="My Date" name="myDate" />
        <Range label="My Range" name="myRange" />
        <Telephone label="Telephone" name="myTelephone" />
        <DateTime label="My DateTime" type="datetime" name="myDateTime" />
        <DateTime label="My Month" type="month" name="myMonth" />
        <DateTime label="My Week" type="week" name="myWeek" />
        <DateTime
          label="My DateTime"
          type="datetime-local"
          name="myDateTimeLocal"
        />
        <DateTime label="My Time" type="time" name="myTime" />
        <DropDown
          label="My Drop Down"
          initialValue="2"
          data-testid="dd1"
          name="myDropDown"
        >
          <OptionGroup label="First Group">
            <Option initialValue="1">First</Option>
            <Option initialValue="2">Second</Option>
            <Option initialValue="3" label="Third" />
          </OptionGroup>
          <OptionGroup label="Second Group">
            <Option initialValue="11">Second First</Option>
            <Option initialValue="12">Second Second</Option>
            <Option initialValue="13" label="Second Third" />
          </OptionGroup>
          <OptionGroup label="Third Group">
            <Option initialValue="21">Third First</Option>
            <Option initialValue="22">Third Second</Option>
            <Option initialValue="23" label="Third Third" />
          </OptionGroup>
        </DropDown>
        <Button>Save</Button>
      </Fieldset>
    </Form>
  )
}
const AppMulti = ({ onSubmit }) => {
  return (
    <Form
      name="testForm"
      onSubmit={e => {
        onSubmit(e)
      }}
    >
      <Fieldset legend="My Form">
        <TextBox label="My Label" name="myTextBox" />
        <CheckBox
          label="My Checkbox"
          value="checkbox value"
          isChecked={false}
          name="myCheckBox"
        />
        <Radio
          label="My Radio"
          name="myRadio"
          isChecked={false}
          value="myRadio1"
        />
        <DateTime label="My Date" name="myDate" />
        <Range label="My Range" name="myRange" />
        <Telephone label="Telephone" name="myTelephone" />
        <DateTime label="My DateTime" type="datetime" name="myDateTime" />
        <DateTime label="My Month" type="month" name="myMonth" />
        <DateTime label="My Week" type="week" name="myWeek" />
        <DateTime
          label="My DateTime"
          type="datetime-local"
          name="myDateTimeLocal"
        />
        <DateTime label="My Time" type="time" name="myTime" />
      </Fieldset>
      <Fieldset legend="My Form part 2">
        <DropDown
          label="My Drop Down"
          initialValue="2"
          data-testid="dd1"
          name="myDropDown"
        >
          <OptionGroup label="First Group">
            <Option initialValue="1">First</Option>
            <Option initialValue="2">Second</Option>
            <Option initialValue="3" label="Third" />
          </OptionGroup>
          <OptionGroup label="Second Group">
            <Option initialValue="11">Second First</Option>
            <Option initialValue="12">Second Second</Option>
            <Option initialValue="13" label="Second Third" />
          </OptionGroup>
          <OptionGroup label="Third Group">
            <Option initialValue="21">Third First</Option>
            <Option initialValue="22">Third Second</Option>
            <Option initialValue="23" label="Third Third" />
          </OptionGroup>
        </DropDown>
        <Button>Save</Button>
      </Fieldset>
    </Form>
  )
}
describe('The form components as a form', () => {
  afterEach(cleanup)
  it('should render and change accordingly', () => {
    const CHANGED_TEXTBOX = 'Changed Value'
    const { container, getByLabelText, getByTestId, getByText } = render(
      <App
        onSubmit={e => {
          expect(e).toMatchSnapshot()
          expect(e.myTextBox).toBe(CHANGED_TEXTBOX)
          expect(e.myDropDown).toBe('22')
        }}
      />
    )
    const textBox = getByLabelText('My Label')
    const dd = getByTestId('dd1')
    const button = getByText('Save')
    expect(container.firstChild).toMatchSnapshot()
    const dropDownValue = dd.value
    expect(dropDownValue).toBe('2')
    fireEvent.change(textBox, { target: { value: CHANGED_TEXTBOX } })
    fireEvent.change(dd, { target: { value: '22' } })
    fireEvent.click(button)
  })
  it('should render the second', () => {
    const { container, getByText } = render(
      <AppMulti
        onSubmit={e => {
          expect(e).toMatchSnapshot()
        }}
      />
    )

    const button = getByText('Save')
    expect(container.firstChild).toMatchSnapshot()

    fireEvent.click(button)
  })
})
