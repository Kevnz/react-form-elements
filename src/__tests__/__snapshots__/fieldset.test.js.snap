// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The form components as a form should render and change accordingly 1`] = `
<form
  class="rfe-form"
  name="testForm"
>
  <fieldset>
    <legend>
      My Form
    </legend>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-1-1"
      >
        My Label
      </label>
      <input
        id="id-1-1"
        name="myTextBox"
        type="text"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-2-1"
      >
        My Checkbox
      </label>
      <input
        id="id-2-1"
        name="myCheckBox"
        type="checkbox"
        value="checkbox value"
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-3-1"
      >
        My Radio
      </label>
      <input
        id="id-3-1"
        name="myRadio"
        type="radio"
        value="myRadio1"
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-4-1"
      >
        My Date
      </label>
      <input
        id="id-4-1"
        name="myDate"
        type="date"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-5-1"
      >
        My Range
      </label>
      <input
        id="id-5-1"
        max="100"
        min="0"
        step="1"
        type="range"
        value="50"
      />
      <span>
        50
      </span>
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-6-1"
      >
        Telephone
      </label>
      <input
        id="id-6-1"
        name="myTelephone"
        type="tel"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-7-1"
      >
        My DateTime
      </label>
      <input
        id="id-7-1"
        name="myDateTime"
        type="datetime"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-8-1"
      >
        My Month
      </label>
      <input
        id="id-8-1"
        name="myMonth"
        type="month"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-9-1"
      >
        My Week
      </label>
      <input
        id="id-9-1"
        name="myWeek"
        type="week"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-10-1"
      >
        My DateTime
      </label>
      <input
        id="id-10-1"
        name="myDateTimeLocal"
        type="datetime-local"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-11-1"
      >
        My Time
      </label>
      <input
        id="id-11-1"
        name="myTime"
        type="time"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-12-1"
      >
        My Drop Down
      </label>
      <select
        data-testid="dd1"
        id="id-12-1"
      >
        <optgroup
          label="First Group"
        >
          <option
            value="1"
          >
            First
          </option>
          <option
            value="2"
          >
            Second
          </option>
          <option
            value="3"
          >
            Third
          </option>
        </optgroup>
        <optgroup
          label="Second Group"
        >
          <option
            value="11"
          >
            Second First
          </option>
          <option
            value="12"
          >
            Second Second
          </option>
          <option
            value="13"
          >
            Second Third
          </option>
        </optgroup>
        <optgroup
          label="Third Group"
        >
          <option
            value="21"
          >
            Third First
          </option>
          <option
            value="22"
          >
            Third Second
          </option>
          <option
            value="23"
          >
            Third Third
          </option>
        </optgroup>
      </select>
    </div>
    <button>
      Save
    </button>
  </fieldset>
</form>
`;

exports[`The form components as a form should render and change accordingly 2`] = `
Object {
  "myCheckBox": null,
  "myDate": "",
  "myDateTime": "",
  "myDateTimeLocal": "",
  "myDropDown": "22",
  "myMonth": "",
  "myRadio": null,
  "myRange": "50",
  "myTelephone": "",
  "myTextBox": "Changed Value",
  "myTime": "",
  "myWeek": "",
}
`;

exports[`The form components as a form should render the second 1`] = `
<form
  class="rfe-form"
  name="testForm"
>
  <fieldset>
    <legend>
      My Form
    </legend>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-15-1"
      >
        My Label
      </label>
      <input
        id="id-15-1"
        name="myTextBox"
        type="text"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-16-1"
      >
        My Checkbox
      </label>
      <input
        id="id-16-1"
        name="myCheckBox"
        type="checkbox"
        value="checkbox value"
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-17-1"
      >
        My Radio
      </label>
      <input
        id="id-17-1"
        name="myRadio"
        type="radio"
        value="myRadio1"
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-18-1"
      >
        My Date
      </label>
      <input
        id="id-18-1"
        name="myDate"
        type="date"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-19-1"
      >
        My Range
      </label>
      <input
        id="id-19-1"
        max="100"
        min="0"
        step="1"
        type="range"
        value="50"
      />
      <span>
        50
      </span>
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-20-1"
      >
        Telephone
      </label>
      <input
        id="id-20-1"
        name="myTelephone"
        type="tel"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-21-1"
      >
        My DateTime
      </label>
      <input
        id="id-21-1"
        name="myDateTime"
        type="datetime"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-22-1"
      >
        My Month
      </label>
      <input
        id="id-22-1"
        name="myMonth"
        type="month"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-23-1"
      >
        My Week
      </label>
      <input
        id="id-23-1"
        name="myWeek"
        type="week"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-24-1"
      >
        My DateTime
      </label>
      <input
        id="id-24-1"
        name="myDateTimeLocal"
        type="datetime-local"
        value=""
      />
    </div>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-25-1"
      >
        My Time
      </label>
      <input
        id="id-25-1"
        name="myTime"
        type="time"
        value=""
      />
    </div>
  </fieldset>
  <fieldset>
    <legend>
      My Form part 2
    </legend>
    <div
      class="rfe-form__row"
    >
      <label
        for="id-26-1"
      >
        My Drop Down
      </label>
      <select
        data-testid="dd1"
        id="id-26-1"
      >
        <optgroup
          label="First Group"
        >
          <option
            value="1"
          >
            First
          </option>
          <option
            value="2"
          >
            Second
          </option>
          <option
            value="3"
          >
            Third
          </option>
        </optgroup>
        <optgroup
          label="Second Group"
        >
          <option
            value="11"
          >
            Second First
          </option>
          <option
            value="12"
          >
            Second Second
          </option>
          <option
            value="13"
          >
            Second Third
          </option>
        </optgroup>
        <optgroup
          label="Third Group"
        >
          <option
            value="21"
          >
            Third First
          </option>
          <option
            value="22"
          >
            Third Second
          </option>
          <option
            value="23"
          >
            Third Third
          </option>
        </optgroup>
      </select>
    </div>
    <button>
      Save
    </button>
  </fieldset>
</form>
`;

exports[`The form components as a form should render the second 2`] = `
Object {
  "myCheckBox": null,
  "myDate": "",
  "myDateTime": "",
  "myDateTimeLocal": "",
  "myDropDown": "2",
  "myMonth": "",
  "myRadio": null,
  "myRange": "50",
  "myTelephone": "",
  "myTextBox": "",
  "myTime": "",
  "myWeek": "",
}
`;
